src/field/vecops.cxx

```diff
@@ -186,8 +187,13 @@ const Field3D Div(const Vector3D& v, CELL_LOC outloc, const std::string& method)
   Vector3D vcn = v;
   vcn.toContravariant();
 
-  Field3D result = DDX(vcn.x.getCoordinates()->J * vcn.x, outloc, method);
-  result += DDY(vcn.y.getCoordinates()->J * vcn.y, outloc, method);
+  Field3D vcnJy = vcn.y.getCoordinates()->J * vcn.y;
+  if (!vcn.y.hasParallelSlices()) {
+    localmesh->communicate(vcnJy);
+  }
```

Comment:
This is actually different from what was in [451037f](https://github.com/boutproject/BOUT-dev/commit/451037f2ef5ee681c8495936d102f6f560ea5615), which I think was more correct there.  I did this because otherwise FCI operators hit an error (currently I forget what it is but I remember talking with @ZedThree about it). 

We should check if vcn has parallel slices.  If it does not, we need to make sure that we get them.  Here's how it looks in [451037f](https://github.com/boutproject/BOUT-dev/commit/451037f2ef5ee681c8495936d102f6f560ea5615): 

`  if(!vcn.y.hasParallelSlices()){
    Field3D vcnJy = vcn.y.getCoordinates()->J * vcn.y;
    localmesh->communicate(vcnJy);
    result = DDY(vcnJy, outloc, method);
  }else{
    result = DDY(vcn.y.getCoordinates()->J * vcn.y, outloc, method);
  }`

Reply:
The branch with the `communicate` looks correct to me, but I'd expect the other one to fail because the `yup`/`ydown` fields (even if they were there on `vcn.y` to start with) are discarded by the multiply operation. Is it possible that you fixed the error just because you only hit the first branch?

I'd think we basically always want the `communicate` call in here. The times when we wouldn't are:
* when using `ParallelTransformIdentity`, but I think this is uncommon and the code with `communicate` would be correct, just slightly sub-optimal so I wouldn't worry about this
* when using `toFieldAligned` and `fromFieldAligned` for parallel derivatives, but all the operators with mixed perpendicular and parallel derivatives are sub-optimal for this case at the moment (because to avoid repeating `toFieldAligned` calls they'd require both a non-aligned and an aligned input), so again I wouldn't worry about a possible extra communicate for the moment.

(you can format a multi-line code block between triple-backticks
\`\`\`
...code...
\`\`\`
)

Reply:
Following discussion with @bshanahan on Slack, we think that the current version is working because `vcn.y` never has parallel slices (they are removed from `v` by `operator=` when `vcn` is created) so the code always goes through the branch with `communicate`, which is always correct but might be sub-optimal. Also, we should only need to calculate parallel slices here not communicate. So I think the thing to do might be:
```suggestion
  if (v.y.hasParallelSlices()) {
    // If v.y has parallel slices then we are using ShiftedMetric (with 
    // mesh:calcParallelSlices_on_communicate=true) or FCI, so we should calculate
    // parallel slices for vcnJy in order to calculate the parallel derivative DDY
    vcnJy.calcParallelSlices();
  }
```